woocommerce/woocommerce-ios
# Shipping Labels: Customize print label screen for purchase flow

Part of #4089

## Description

This customizes the `Print Shipping Label` screen so it can be used in the shipping label purchase flow (in addition to reprinting existing shipping labels).

This PR also hooks up the "Mark this order complete and notify the customer" toggle in the Shipping Label Form, so the order is marked complete when that toggle is enabled.

## Changes

* Refactors the `Print Shipping Label` view controller, view model, and coordinator to use a more general name ("print" instead of "reprint").
* Adds a `PrintType` to `PrintShippingLabelCoordinator`, to handle whether it is printing a new label or reprinting an existing label.
* Customizes `PrintShippingLabelViewController` based on the `PrintType`. The screen doesn't change when reprinting a label, but when printing a label it shows the relevant header text, header image, and "Save for Later" button in addition to the print options.
* Removes the `Shipping Label Form` screen from the navigation stack, so when you tap the back button on the `Print Shipping Label` screen you go back to the order details.
* Hooks up the "Mark this order complete and notify the customer" toggle in the `Shipping Label Form` screen. When the label purchase is successful, if this toggle is enabled the order is marked as complete.

Reprint shipping label (existing labels) | Print shipping label (new purchase) | Print shipping label (after selecting paper size)
-|-|-
![Simulator Screen Shot - iPhone 11 Pro - 2021-07-06 at 15 07 25](https://user-images.githubusercontent.com/8658164/124616754-42000b00-de6e-11eb-896a-9178a10b693b.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-07-06 at 15 08 16](https://user-images.githubusercontent.com/8658164/124616773-45939200-de6e-11eb-8787-ffb2a4a9cddd.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-07-06 at 15 08 23](https://user-images.githubusercontent.com/8658164/124616783-475d5580-de6e-11eb-927a-b36884dff3ef.png)


## Testing

1. Make sure you have configured the WooCommerce WCShip plugin, and you have one or more payment methods set up in WP Admin under WooCommerce > Settings > Shipping > WooCommerce Shipping.
2. Launch the app in debug mode (because of the feature flag).
3. Open an order detail that is eligible for the shipping label.
4. Tap the button "Create Shipping Label".
5. Complete all the fields in the form.
6. In the Order Summary section, enable the "Mark order complete" toggle and tap the button "Purchase Label".
7. On the Print Shipping Label screen, select the paper size and confirm the "Print Shipping Label" button is enabled and you can tap it to print your label.
8. Confirm you can tap the help text ("See layout ..." and "Don't know how ...") for more information.
9. Confirm you can tap the "Save for Later" button and you are taken back to the order details, which updates to display your purchased shipping label and shows the order with a `Completed` status.

Additional testing:

* Repeat the steps above, but don't enable the "Mark order complete" toggle and confirm your order is not marked as complete.
* Repeat the steps above, but tap the Back button on the Print Shipping Label screen and confirm you are taken back to the order details (not the Shipping Label Form).

## Submitter Checklist

Update release notes:
